src/Content/Backend/Solution/Monaco.Template.Backend.ArchitectureTests/ApplicationTests.cs

```diff
@@ -0,0 +1,162 @@
+						  .AndShould()
+						  .HaveName("Command")
+						  .OrShould()
+						  .HaveName("Query")
+						  .Check(Architecture);
+
+	[Fact(DisplayName = "Commands have validators")]
+	public void CommandsHaveValidators() =>
+		_commands.Should()
+				 .FollowCustomCondition(command => _validators.GetObjects(Architecture)
+															  .Any(v => v.GetInheritsBaseClassDependencies()
+																		 .Any(d => d.Target.Equals(Validator) &&
+																				   d.TargetGenericArguments
+																					.Any(g => g.Type.Equals(command)))),
+										"have a validator",
+										"does not have a validator")
+				 .Check(Architecture);
+
+	[Fact(DisplayName = "Validators exist in Application layer and are sealed, internal and nested")]
+	public void ValidatorsExistInApplicationAreSealedInternalAndNested() =>
+		_validators.Should()
+				   .Be(_applicationLayer)
+				   .AndShould()
+				   .HaveName("Validator")
+				   .AndShould()
+				   .BeSealed()
+				   .AndShould()
+				   .BeInternal()
+				   .AndShould()
+				   .BeNested()
+				   .Check(Architecture);
+
+	[Fact(DisplayName = "Handlers exist in Application layer, are named Handler and are nested, sealed and internal")]
+	public void HandlersExistInApplicationAreNamedHandlerAndAreNestedSealedAndInternal() =>
+		_handlers.Should()
+				 .Be(_applicationLayer)
+				 .AndShould()
+				 .HaveName("Handler")
+				 .AndShould()
+				 .BeNested()
+				 .AndShould()
+				 .BeSealed()
+				 .AndShould()
+				 .BeInternal()
+				 .Check(Architecture);
+
+	[Fact(DisplayName = "Command, validator and handler should be nested together")]
+	public void CommandValidatorAndHandlerAreNestedTogether() =>
+		_commands.Should()
+				 .FollowCustomCondition(c => _validators.GetObjects(Architecture)
+														.Single(v => v.GetInheritsBaseClassDependencies()
+																	  .Any(d => d.Target.Equals(Validator) &&
+																				d.TargetGenericArguments
+																				 .Any(g => g.Type.Equals(c))))
+														.NestType(Architecture)!
+														.Equals(c.NestType(Architecture)),
+										"be nested together with their validator",
+										"is not nested together with its validator")
+				 .AndShould()
+				 .FollowCustomCondition(c => _handlers.GetObjects(Architecture)
+													  .Single(h => h.GetImplementsInterfaceDependencies()
+																	.Any(i => i.Target.Equals(RequestHandlerReturn) &&
+																			  i.TargetGenericArguments
+																			   .Any(g => g.Type.Equals(c))))
+													  .NestType(Architecture)!
+													  .Equals(c.NestType(Architecture)),
+										"be nested together with their handler",
+										"is not nested together with its handler")
+				 .Check(Architecture);
+
+	[Fact(DisplayName = "Features are free of cycles")]
+	public void FeaturesAreFreeOfCycles()
+	{
+		Slices().Matching("Monaco.Template.Backend.Application.Features.(*)")
+				.Should()
+				.BeFreeOfCycles()
+				.Evaluate(Architecture);
```

Comment:
Should this be using `Check(Architecture)` or `Evaluate(Architecture)`?

Same with the test below.

Reply:
At this point it only offers the Evaluate() method, not the Check() one.

Reply:
Oh? Maye I'm misunderstanding but I'm able to swap out Evaluate() for Check() with no issues locally.

I think the Evaluate method will only return a `EvaluationResult` objects so we would also need to assert that those objects all have `Passed == true`?

If I change this locally then the second of these 2 tests fails. The Check() version of this provides a nicer to understand error message:
![image](https://github.com/user-attachments/assets/684e6a58-1e75-4d00-9987-bfb0422a5a65)

Reply:
Hmm... 🤔 So weird, I could have sworn that I didn't see it the first time I wrote that part of the code... Seems to be available now, apparently.

But I can't seem to figure a way to exclude the `Product.Extensions` path as a slice to check, so it errors there.
Have you figured a way? ArchUnitNET docs on this feature seem to be kind of scarce and I don't understand quite well how to use those expressions.

Reply:
Yeah, thinking the same thing!!
I have considered option 2 as well, but like you said: we shouldn't need to modify app code just because some test.

Perhaps we need to consider an alternative way to perform this test...

Reply:
@matthewtoghill please check the latest changes. I've replaced the test that was failing by another that explicitly checks no `ISender` is being injected to potentially trigger a Command or Query from a handler.

Failure should be seen like this:
![image](https://github.com/user-attachments/assets/8f1fbe4a-2b8d-4cb0-8945-655696673f63)

Reply:
Makes sense and seems like a sensible alternative way to test whether slices are independent of each other.